### PR TITLE
Add not null values

### PR DESCRIPTION
**Pull request by chcholman**
This is the next PR to come out of our conversation around cleaning up the database schema in May. It includes dropping two unused tables `tasks` and `student_academic_histories`, removing unused fields where possible, and adding not_null values to appropriate fields. All fields where not_null has been added have been checked against the production db to confirm there are no current offending objects.

### CODE REVIEW

**Comment by jwright** [comment]
@chcholman I am unsure why these changes were made. Can you give me some background or context around this PR?

File: app/models/grade.rb [hunk 1]
```diff
@@ -10,8 +10,7 @@ class Grade < ActiveRecord::Base
   belongs_to :student, class_name: "User", touch: true
   belongs_to :team, touch: true
   belongs_to :submission
-  belongs_to :task, touch: true # Optional
-  belongs_to :group, polymorphic: true, touch: true # Optional
```
Comment (jwright) [inline app/models/grade.rb]: 👍  Awesome to see this unnecessary polymorphic `group` go away

File: db/migrate/20160815135612_remove_character_limits.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    change_column :submission_files, :filename, :string, limit: nil
+    change_column :submission_files, :filepath, :string, limit: nil
+    change_column :submission_files, :file, :string, limit: nil
+    
+    change_column :submissions, :feedback, :string, limit: nil
+    change_column :submissions, :comment, :string, limit: nil
+    change_column :submissions, :link, :string, limit: nil
+    change_column :submissions, :assignment_type, :string, limit: nil
+    
+    drop_table :tasks
```
Comment (jwright) [inline db/migrate/20160815135612_remove_character_limits.rb]: It's odd that this is in the `remove_character_limits` migration. Would it be easy/worth it to move it to it's own migration?

File: db/migrate/20160815135612_remove_character_limits.rb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    
+    change_column :teams, :name, :string, limit: nil
+    change_column :teams, :banner, :string, limit: nil
+    
+    drop_table :themes
```
Comment (jwright) [inline db/migrate/20160815135612_remove_character_limits.rb]: Same comment as above

File: spec/controllers/criteria_controller_spec.rb [hunk 4]
```diff
@@ -17,8 +17,11 @@
 
     describe "POST create" do
       it "creates a new criterion" do
-        params = attributes_for(:criterion)
-        expect{ post :create, criterion: params }.to change(Criterion,:count).by(1)
+        @assignment = create(:assignment, course: @course)
```
Comment (jwright) [inline spec/controllers/criteria_controller_spec.rb]: Do these (`@assignment` and `@rubric`) need to be instance variables?

**Comment by jwright** [comment]
@chcholman This looks like a *great* cleanup!

I added a few minor comments but no major things.

File: spec/controllers/criteria_controller_spec.rb (hunk 4 above)
Comment (chcholman) [inline spec/controllers/criteria_controller_spec.rb]: Nope!

**Pull request merged**